### ISSUE TITLE
Error at compiling

### ISSUE THREAD

**Issue by camilo1729** [cea-hpc/wi4mpi#6]
Hello,

I have a problem when trying to compile following the instructions, I do the following:

```bash
$ mkdir BUILD ; cd BUILD
cr3db69n@eofront5:/home/cristian/repositories/wi4mpi/BUILD$ cmake -DCMAKE_INSTALL_PREFIX=/home/cristian/software/wi4mpi/ /home/cristian/repositories/wi4mpi/
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- The Fortran compiler identification is GNU
-- Check for working Fortran compiler: /usr/bin/gfortran
-- Check for working Fortran compiler: /usr/bin/gfortran  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/bin/gfortran supports Fortran 90
-- Checking whether /usr/bin/gfortran supports Fortran 90 -- yes
-- Found MPI_C: /usr/lib/libmpi.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libhwloc.so
-- Found MPI_CXX: /usr/lib/libmpi_cxx.so;/usr/lib/libmpi.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libhwloc.so
-- Found MPI_Fortran: /usr/lib/libmpi_f90.so;/usr/lib/libmpi_f77.so;/usr/lib/libmpi.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libhwloc.so
Make Error at CMakeLists.txt:90 (get_filename_component):
  get_filename_component unknown component /usr/lib/openmpi/include/openmpi


-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
status/gpfseole/home/cr3db69n/repositories/wi4mpi/BUILD/src
-- Configuring incomplete, errors occurred!
See also "/home/cristian/repositories/wi4mpi/BUILD/CMakeFiles/CMakeOutput.log".
See also "/home/cristian/repositories/wi4mpi/BUILD/CMakeFiles/CMakeError.log".

```

Any ideas why is that happening? 

Thank you in advance

**Comment by camilo1729** [comment]
I look further on the previous error and it seemed that it  comes from this line in CMakelist
```
get_filename_component(OPENMPI_ROOT ${MPI_C_INCLUDE_PATH} DIRECTORY)

CMake Error at CMakeLists.txt:90 (get_filename_component):
  get_filename_component unknown component
  /opt/openmpi-1.10.4/include/openmpi/opal/mca/event/libevent2021/libevent/include
```
When I commented out this line the process finished successfully
```
Scanning dependencies of target mpi
[  1%] Building C object src/CMakeFiles/mpi.dir/interface/gen/interface_test.c.o
[  2%] Building C object src/CMakeFiles/mpi.dir/interface/gen/interface_fort.c.o
In file included from /home/cr3db69n/repositories/wi4mpi/src/interface/gen/interface_fort.c:30:0:
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:44:8: error: conflicting types for ‘INTERFACE_F_LOCAL_MPI_Get_processor_name’
 void (*INTERFACE_F_LOCAL_MPI_Get_processor_name)(char *,int,int *,int *);
        ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:40:8: note: previous declaration of ‘INTERFACE_F_LOCAL_MPI_Get_processor_name’ was here
 void (*INTERFACE_F_LOCAL_MPI_Get_processor_name)(char *,int *,int *,int);
        ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:49:6: error: conflicting types for ‘pmpi_get_processor_name_’
 void pmpi_get_processor_name_(char * name,int namelen,int * resultlen,int * ret)
      ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:32:6: note: previous declaration of ‘pmpi_get_processor_name_’ was here
 void pmpi_get_processor_name_(char *,int *,int *,int);
      ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:94:9: error: conflicting types for ‘INTERFACE_F_LOCAL_MPI_Error_string’
 void  (*INTERFACE_F_LOCAL_MPI_Error_string)(int * errorcode,int reslen,char * string,int * resultlen,int * ret);
         ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:90:9: note: previous declaration of ‘INTERFACE_F_LOCAL_MPI_Error_string’ was here
 void  (*INTERFACE_F_LOCAL_MPI_Error_string)(int * errorcode,char * string,int * resultlen,int * ret,int reslen);
         ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h: In function ‘pmpi_error_string_’:
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:109:50: warning: passing argument 2 of ‘INTERFACE_F_LOCAL_MPI_Error_string’ makes integer from pointer without a cast
     INTERFACE_F_LOCAL_MPI_Error_string(errorcode,string,reslen,resultlen,ret);
                                                  ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:109:50: note: expected ‘int’ but argument is of type ‘char *’
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:109:57: warning: passing argument 3 of ‘INTERFACE_F_LOCAL_MPI_Error_string’ makes pointer from integer without a cast
     INTERFACE_F_LOCAL_MPI_Error_string(errorcode,string,reslen,resultlen,ret);
                                                         ^
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:109:57: note: expected ‘char *’ but argument is of type ‘int’
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h: At top level:
/home/cr3db69n/repositories/wi4mpi/src/interface/manual/manual_interface.h:149:8: error: conflicting types for ‘INTERFACE_F_LOCAL_MPI_File_open’
 void (*INTERFACE_F_LOCAL_MPI_File_open)(int *, char*, int, int*, int*,int*, int*);
```

But I have errors about conflicting types. Do you have any idea about why my first cmake failed with in that specific line in the CMakeList.  I tried different version of OpenMPI without success.

**Comment by adrien-cotte** [comment]
Hello,

This is quite old but the patch proposed here should work fine:
https://github.com/cea-hpc/wi4mpi/issues/8

**Comment by marcjoos-cea** [comment]
This issue must have been fixed by PR#8. Please reopen the issue is the problem reappears.
